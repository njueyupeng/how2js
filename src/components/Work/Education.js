import React from 'react'
import PropTypes from 'prop-types'

const Education = (props) => {
  const { theme, school, degree, gpa, honors } = props

  return (
    <React.Fragment>
      <div className="education">
        <div className="schoolInfo">
          <div className="school">{school}</div>
          <div className="degree">{degree}</div>
        </div>
        <div className="studentInfo">
          <div>GPA: {gpa}</div>
          {honors && <div>{honors}</div>}
        </div>
      </div>
      {/* --- STYLES --- */}
      <style jsx>{`
        .education {
          padding-bottom: 15px;
        }
        .schoolInfo {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          justify-content: space-between;
          align-content: center;
        }
        .studentInfo {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          align-content: center;
          justify-content: space-between;
        }
        .school {
          font-size: 1.2rem;
          line-height: 1.42857;
          font-weight: 600;
          margin-bottom: 2px;
        }
        .degree {
          font-size: 1rem;
          color: ${theme.color.neutral.gray.g};
          font-weight: 550;
        }
      `}</style>
    </React.Fragment>
  )
}

Education.propTypes = {
  theme: PropTypes.object.isRequired,
  school: PropTypes.string.isRequired,
  degree: PropTypes.string.isRequired,
  gpa: PropTypes.string.isRequired,
  honors: PropTypes.string,
}

export default Education
