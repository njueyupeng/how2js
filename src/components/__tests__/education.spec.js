import React from 'react'
import { create } from 'react-test-renderer'
import Education from '../Work/Education'
const theme = require('../../theme/theme.json')

describe('Education Component', () => {
  it('renders correctly without honors', () => {
    const tree = create(
      <Education
        theme={theme}
        school="Milwaukee Area Technical College"
        degree="A.S. Mobile Application Development"
        gpa="3.729"
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly with honors', () => {
    const tree = create(
      <Education
        theme={theme}
        school="Lakeland University"
        degree="B.S. Computer Science"
        gpa="3.9375"
        honors="Summa cum Laude"
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
