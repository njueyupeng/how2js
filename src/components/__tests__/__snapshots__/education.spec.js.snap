// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education Component renders correctly with honors 1`] = `
Array [
  <div
    className="education"
  >
    <div
      className="schoolInfo"
    >
      <div
        className="school"
      >
        Lakeland University
      </div>
      <div
        className="degree"
      >
        B.S. Computer Science
      </div>
    </div>
    <div
      className="studentInfo"
    >
      <div>
        GPA: 
        3.9375
      </div>
      <div>
        Summa cum Laude
      </div>
    </div>
  </div>,
  <style
    jsx={true}
  >
    
        .education {
          padding-bottom: 15px;
        }
        .schoolInfo {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          justify-content: space-between;
          align-content: center;
        }
        .studentInfo {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          align-content: center;
          justify-content: space-between;
        }
        .school {
          font-size: 1.2rem;
          line-height: 1.42857;
          font-weight: 600;
          margin-bottom: 2px;
        }
        .degree {
          font-size: 1rem;
          color: #2c3e50;
          font-weight: 550;
        }
      
  </style>,
]
`;

exports[`Education Component renders correctly without honors 1`] = `
Array [
  <div
    className="education"
  >
    <div
      className="schoolInfo"
    >
      <div
        className="school"
      >
        Milwaukee Area Technical College
      </div>
      <div
        className="degree"
      >
        A.S. Mobile Application Development
      </div>
    </div>
    <div
      className="studentInfo"
    >
      <div>
        GPA: 
        3.729
      </div>
    </div>
  </div>,
  <style
    jsx={true}
  >
    
        .education {
          padding-bottom: 15px;
        }
        .schoolInfo {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          justify-content: space-between;
          align-content: center;
        }
        .studentInfo {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          align-content: center;
          justify-content: space-between;
        }
        .school {
          font-size: 1.2rem;
          line-height: 1.42857;
          font-weight: 600;
          margin-bottom: 2px;
        }
        .degree {
          font-size: 1rem;
          color: #2c3e50;
          font-weight: 550;
        }
      
  </style>,
]
`;
